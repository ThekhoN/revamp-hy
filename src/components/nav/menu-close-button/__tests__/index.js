import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import MenuCloseButton from '../index';
import sinon from 'sinon';

describe('MenuCloseButton', () => {
	// snapshot test
	it('renders correctly', () => {
		const tree = renderer.create(<MenuCloseButton />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});

/*
describe('MenuCloseButton', () => {
  let state = 'open';
  const closeFunction= () => {
    state = 'close';
  }
  it('runs function on click', () => {
    expect(state).toEqual('open');
    const wrapper = shallow(<MenuCloseButton />)
    wrapper.find('div').simulate('click', {
      closeFunction(){}
    });
    expect(state).toEqual('close'); // fail
  });
});
*/

describe('MenuCloseButton', () => {
	it('triggersClickEvent on click', () => {
		const handleOnClickEvent = sinon.spy();
		const wrapper = shallow(
			<MenuCloseButton handleOnClick={handleOnClickEvent} />
		);
		wrapper.find('div').simulate('click');
		expect(handleOnClickEvent.calledOnce).toEqual(true);
	});
});

/* https://github.com/airbnb/enzyme/blob/master/docs/api/shallow.md */
