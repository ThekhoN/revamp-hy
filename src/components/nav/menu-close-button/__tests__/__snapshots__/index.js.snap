// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuCloseButton renders correctly 1`] = `
<div
  className="menu-close-button mobile-ui--only align--top-right--abs"
  onClick={undefined}
>
  <span
    className="svg-icon-wrapper"
  >
    <svg
      aria-labelledby="title"
      height="48"
      viewBox="0 0 48 48"
      width="48"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title
        id="title"
      >
        Homeyantra Navigation Close
      </title>
      <path
        d="M25.4 24l4.9-4.9c.4-.4.5-1.1.1-1.5-.4-.4-1.1-.4-1.5.1L24 22.6l-4.9-4.9c-.4-.4-1.1-.5-1.5-.1-.4.4-.4 1.1.1 1.5l4.9 4.9-4.9 4.9c-.4.4-.5 1.1-.1 1.5s1.1.4 1.5-.1l4.9-4.9 4.9 4.9c.4.4 1.1.5 1.5.1s.4-1.1-.1-1.5L25.4 24z"
        fill="#38afc7"
      />
    </svg>
  </span>
</div>
`;
